File: go/internal/agent/lintfixer/lint.go
// Package lintfixer is the lint-remediation configuration of the fixflow engine: it
// supplies a triage step (normalize a linter report) and an analyze step (rewrite
// the affected source files), plus its branch/label/check identity. The loop itself
// lives in internal/agent/fixflow.
package lintfixer

import (
	"embed"

	"github.com/jkjamies/automation-agent/internal/agent/fixflow"
	"github.com/jkjamies/automation-agent/internal/agent/setup"
)

//go:embed prompts/*.md
var promptFS embed.FS

var prompts = setup.NewPrompts(promptFS)

// NewEngine builds the lint-fixer engine.
func NewEngine(d fixflow.Deps) *fixflow.Engine {
	return fixflow.NewEngine(fixflow.Spec{
		Name:          "lint",
		Branch:        "automation-agent/lint-fix",
		Label:         "automation-agent",
		CheckName:     "agent-lint-verify",
		CommitMessage: "automation-agent: fix lint problems",
		PRTitle:       "automation-agent: fix lint problems",
		SuccessTitle:  "Lint fix succeeded ✅",
		ReviewTitle:   "Lint fix needs human review ⚠️",
		Triage:        Triage,
		Analyze:       Analyze,
	}, d)
}

File: kotlin/src/main/kotlin/com/automation/agent/agent/lintfixer/Lint.kt
/*
 * Package lintfixer is the lint-remediation configuration of the fixflow engine: it supplies a
 * triage step (normalize a linter report) and an analyze step (rewrite the affected source files),
 * plus its branch/label/check identity. The loop itself lives in agent.fixflow.
 */
package com.automation.agent.agent.lintfixer

import com.automation.agent.agent.fixflow.Deps
import com.automation.agent.agent.fixflow.Engine
import com.automation.agent.agent.fixflow.Spec
import com.automation.agent.agent.setup.Prompts

internal val prompts = Prompts.forAgent("lintfixer")

/** Builds the lint-fixer engine. */
fun newEngine(deps: Deps): Engine =
    Engine(
        Spec(
            name = "lint",
            branch = "automation-agent/lint-fix",
            checkName = "agent-lint-verify",
            commitMessage = "automation-agent: fix lint problems",
            prTitle = "automation-agent: fix lint problems",
            successTitle = "Lint fix succeeded ✅",
            reviewTitle = "Lint fix needs human review ⚠️",
            cleanTitle = "Lint already clean 👏",
            triage = ::triage,
            analyze = ::analyze,
        ),
        deps,
    )
